app/src/main/java/kth/jjve/xfran/ProfileActivity.java

```diff
@@ -1,14 +1,123 @@
 package kth.jjve.xfran;
 
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.ActionBarDrawerToggle;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.view.GravityCompat;
+import androidx.drawerlayout.widget.DrawerLayout;
+import androidx.lifecycle.ViewModelProviders;
 
+import android.annotation.SuppressLint;
+import android.content.Intent;
 import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.ImageButton;
+import android.widget.TextView;
+import android.widget.Toast;
 
-public class ProfileActivity extends AppCompatActivity {
+import com.google.android.material.navigation.NavigationView;
 
+import kth.jjve.xfran.models.UserProfile;
+import kth.jjve.xfran.viewmodels.UserProfileVM;
+
+public class ProfileActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {
+    /*_________ VIEW _________*/
+    private TextView mName, mEmail, mWeight, mHeight;
+    private DrawerLayout drawerLayout;
+    private NavigationView navigationView;
+    private Toolbar toolbar;
+    private Menu profileMenu;
+
+    @SuppressLint("SetTextI18n")
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_profile);
+
+        /*----- HOOKS -----*/
+        drawerLayout = findViewById(R.id.drawer_layout_profile);
+        navigationView = findViewById(R.id.nav_view_profile);
+        toolbar = findViewById(R.id.profile_toolbar);
+
+        mName = findViewById(R.id.profile_username);
+        mEmail = findViewById(R.id.profile_useremail);
+        mWeight = findViewById(R.id.profile_weight);
+        mHeight = findViewById(R.id.profile_height);
+        ImageButton edit = findViewById(R.id.profile_Edit);
+
+        /*-----  VM  -----*/
+        UserProfileVM mUserProfileVM = ViewModelProviders.of(this).get(UserProfileVM.class);
+        mUserProfileVM.init();
+        mUserProfileVM.getUserProfile().observe(this, this::setViews);
+
+        /*--- OBSERVER ---*/
+        edit.setOnClickListener(v -> startActivity(new Intent(ProfileActivity.this, EditProfileActivity.class)));
+
+        /*----- INIT -----*/
+        setSupportActionBar(toolbar);
+        initNavMenu();
+
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        navigationView.setCheckedItem(R.id.nav_profile);
```

Comment:
When you update an info on edit profile, it doesnt show the new info on the profile activity after you save it. Thats because it runs on resume and not on create when you "come back" from edit profile. So the updated info will only appear if you change to another activity and come back. Maybe put something on resume to update info (or after saving).

Reply:
will do later